src/model/pwrmgmt.go

```diff
@@ -23,15 +25,76 @@ var scalProfilesMap = map[string]ScalProfiles{
 type CpuGovernanceRule struct {
 	CPUs    string `yaml:"cpus"`
 	ScalGov string `yaml:"scaling_governor"`
+	MinFreq string `yaml:"min_freq"`
+	MaxFreq string `yaml:"max_freq"`
 }
 
 func (c CpuGovernanceRule) Validate() error {
-	if _, ok := scalProfilesMap[c.ScalGov]; !ok {
+	if _, err := cpulists.Parse(c.CPUs); err != nil {
+		return err
+	}
+	if _, ok := scalProfilesMap[c.ScalGov]; !ok && c.ScalGov != "" {
 		return fmt.Errorf("invalid cpu scaling governor: %v", c.ScalGov)
 	}
-	_, err := cpulists.Parse(c.CPUs)
-	if err != nil {
-		return fmt.Errorf("invalid cpus: %v", err)
+	min, errMin := ParseFreq(c.MinFreq)
+	if errMin != nil {
+		return fmt.Errorf("invalid min_freq: %v", errMin)
+	}
+	max, errMax := ParseFreq(c.MaxFreq)
+	if errMax != nil {
+		return fmt.Errorf("invalid max_freq: %v", errMax)
+	}
+
+	minAndmaxAreSet := (min != -1 && max != -1) && (min != 0 && max != 0)
```

Comment:
1. Mismatching naming convention. But it doesn't look like this variable is even needed. The condition can be checked in an if statement before scoping the remaining checks.
2. Why checking (min != 0 && max != 0) ?